src/dune_engine/scheduler.ml

```diff
@@ -978,11 +977,5 @@ let shutdown () =
   fill_file_changes
 
 let csexp_scheduler () =
-  let t = Fiber.Var.get_exn t_var in
+  let+ t = Fiber.Var.get_exn t_var in
   Lazy.force t.csexp_scheduler
-
-let add_to_env env =
```

Comment:
Don't we still need this? Otherwise child processes will not know how to connect to RPC.

Reply:
Indeed. I initially removed as because it spreads the monad in `default_env` in `process.ml`, which makes the lazy useless. However, now that I think about it, `default_env` is pretty much never used given that we always use the context environment. Which also means that the `unix` field of `Env.t` is currently useless...

We should probably move this environment extensions somewhere more central. I've restored it for now, but moved the function to the `Config` sub-module since that's all we need and we already a `Scheduler.Config.t` at had in `run_internal`.